db/migrate/20230126233821_delete_calculator_currency_from_spree_preferences.rb

```diff
@@ -0,0 +1,15 @@
+class DeleteCalculatorCurrencyFromSpreePreferences < ActiveRecord::Migration[6.1]
+  class SpreePreference < ActiveRecord::Base
+  end
+
+  def up
+    # Delete all currency preferences for calculators, they are no longer used.
+    rows = SpreePreference.where("key like '/calculator%/currency/%'").delete_all
+    puts "#{rows} rows deleted."
+  end
+
+  def down
+    # Sorry, the data was deleted!
+    raise ActiveRecord::IrreversibleMigration
+  end
```

Comment:
I often wonder what's best practice here. I tend to not raise an error. So if I want to repeat it, I can rollback and migrate without problem. Or maybe you want to roll back to an older migration and don't need this.

It doesn't really matter, I'm bike-shedding here, but I think that when the rollback doesn't do any harm either then we don't need to raise. If you did raise though then you can pass a message to the error instead of writing a comment. Or do a puts to explain why we can't reverse.

Reply:
That's a good point too. Because I'm here now, I'll update it.